libs/ui-lib/lib/cim/components/common/CpuArchitectureDropdown.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { Dropdown, DropdownItem, DropdownToggle, FormGroup } from '@patternfly/react-core';
+import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
+import {
+  architectureData,
+  ClusterDetailsValues,
+  CpuArchitecture,
+  getFieldId,
+  StaticField,
+  SupportedCpuArchitecture,
+} from '../../../common';
+import { useField, useFormikContext } from 'formik';
+
+const CpuArchitectureDropdown = ({ isDisabled = false }: { isDisabled?: boolean }) => {
+  const { t } = useTranslation();
+  const { setFieldValue } = useFormikContext<ClusterDetailsValues>();
+  const [{ name, value }, , { setValue }] = useField<SupportedCpuArchitecture>('cpuArchitecture');
+  const [cpuArchOpen, setCpuArchOpen] = React.useState(false);
+  const fieldId = getFieldId(name, 'input');
+
+  const onCpuArchToggle = React.useCallback(() => {
+    if (cpuArchOpen) {
+      setCpuArchOpen(false);
+    } else {
+      setCpuArchOpen(true);
+    }
+  }, [cpuArchOpen]);
+
+  const onCpuArchSelect = React.useCallback(
+    (e?: React.SyntheticEvent<HTMLDivElement>) => {
+      const val = e?.currentTarget.id as SupportedCpuArchitecture;
+      setValue(val);
+
+      if (val === CpuArchitecture.s390x) {
+        setFieldValue('userManagedNetworking', true);
+      } else {
+        setFieldValue('userManagedNetworking', false);
+      }
+      setCpuArchOpen(false);
+    },
+    [setValue, setFieldValue],
+  );
+
```

Comment:
Again, no need to memoize. 

There is also a bug in formik, where `setFieldValue` is always a new instance - we have a workaround for that in [useFormikHelpers.ts](https://github.com/openshift-assisted/assisted-installer-ui/blob/master/libs/ui-lib/lib/common/hooks/useFormikHelpers.ts)

We can also simplify `setFieldValue('userManagedNetworking', val === CpuArchitecture.s390x);`